lib/iob/calculate.js

```diff
@@ -37,4 +58,76 @@ function iobCalc(treatment, time, dia) {
     return results;
 }
 
-exports = module.exports = iobCalc;
+
+function iobCalc(treatment, time, dia, profile) {
+
+	var curve = profile.curve;
+
+    if (curve === undefined || curve.toLowerCase() == 'bilinear') {
+    	return iobCalcBiLinear(treatment, time, dia);
+    };
+    
+    curve = curve.toLowerCase();
+    
+    if (curve != 'novorapid' && curve != 'fiasp') {
+    	console.error('Unsupported curve function: "' + curve + '". Supported curves: bilinear, novorapid and fiasp.');
+    	return;
+    }
+
+	var usePeakTime = false;
+
+	if (profile.dia !== 5 && profile.insulinPeakTime !== undefined) {
+		console.error('Pump DIA must be set to 5 hours if using the Insulin Peak Time setting');
+		
+		if (profile.insulinPeakTime < 35 || profile.insulinPeakTime > 80) {
+			console.error('Insulin Peak Time is only supported for values between 35 to 80 minutes');
+		}
+		return;
+	}
+	
+	if (profile.insulinPeakTime !== undefined) {
+		usePeakTime = true;
+	}
+
+	// default to the published Fiasp DIA
+
+    var diaratio =  5.0 / profile.dia;
+
+	if (profile.insulinPeakTime !== undefined) {
+		diaratio = 55.0 / Number(profile.insulinPeakTime);
+	}
+    
+    var end = 300;
+
+    if (typeof time === 'undefined') {
+        time = new Date();
+    }
+
+    var results = {};
+
+    if (treatment.insulin) {
+        var bolusTime = new Date(treatment.date);
+        var minAgo = diaratio * (time - bolusTime) / 1000 / 60;
```

Comment:
If you have the spreadsheet up, I'd like to see how the peak time adjustments affect NovoRapid too.  This looks to me like it will completely break as it currently stands.